src/OptimizeImageTrait.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/tiny-png-sonata-media-bundle project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\TinyPngSonataMediaBundle;
+
+use Ekino\TinyPngSonataMediaBundle\Consumer\OptimizeImageConsumer;
+use Sonata\NotificationBundle\Backend\BackendInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Trait OptimizeImageTrait.
+ *
+ * @author William JEHANNE <william.jehanne@ekino.com>
+ */
+trait OptimizeImageTrait
+{
+    /**
+     * @var BackendInterface
+     */
+    private $backend;
+
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param Request|null $request
+     *
+     * @return RedirectResponse
+     */
+    public function optimizeAction(Request $request = null): RedirectResponse
+    {
+        $this->backend->createAndPublish(OptimizeImageConsumer::CONSUMER_TYPE, [
+            'mediaId' => $request->get('id'),
+        ]);
+
+        $this->session->getFlashBag()->add(
+            'sonata_flash_success',
+            $this->translator->trans('optimize.image.success', [], 'TinyPngSonataMediaBundle')
+        );
+
+        return $this->redirectToList();
+    }
+
+    /**
+     * @param BackendInterface $backend
+     */
+    public function setBackend(BackendInterface $backend): void
```

Comment:
why can't we use autowiring ?

Reply:
no auto wiring in third party bundles ;) never ;) https://symfony.com/doc/current/service_container/autowiring.html#public-and-reusable-bundles